app/actions/remote/post.ts

```diff
@@ -935,6 +939,17 @@ export const editPost = async (serverUrl: string, postId: string, postMessage: s
                     await operator.batchRecords(fileDeleteModels, 'delete files');
                 }
             }
+
+            const filesChanged = originalFileIds.length !== file_ids.length ||
+                                (() => {
+                                    const fileIdsSet = new Set(file_ids);
+                                    return !originalFileIds.every((id) => fileIdsSet.has(id));
+                                })();
```

Comment:
Move this logic to a util function. You are using it here and in the websocket.

Also, let's not use the `(() => foo)()` pattern. It is quite confusing :sweat_smile:

We can do something like:

```
function hasFilesChanged(a: string[], b: string[]) {
    if (a.length !== b.length) {
        return true;
    }
    const fileIdsSet = new Set(a);
    return !b.every((id) => fileIdsSet.has(id));
}
```

Reply:
Done.